co-simulation/fed/mosaic-infrastructure/src/main/java/org/eclipse/mosaic/fed/infrastructure/ambassador/InfrastructureTimeInterface.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (C) 2023 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.eclipse.mosaic.fed.infrastructure.ambassador;
+import java.io.IOException;
+
+import org.apache.commons.logging.Log;
+
+import com.google.gson.Gson;
+public class InfrastructureTimeInterface{
+public InfrastructureInstanceManager manager;
+//private int target_port=12123;
+//set to false on init release
+private Boolean await_infrastructure_advance_request = false;
+public InfrastructureTimeInterface(InfrastructureInstanceManager manager){
+    this.manager = manager;
+}
+
+/**
+ * This function implements the encoding of a json string consist of timestep and seq
+ * 
+ * @param message This is the class that gets encoded, which include the data of timestep and seq
+ * @return return_json: encoded json string
+ */
+public byte[] encodeTimeMessage(InfrastructureTimeMessage message)
+{
+    Gson gson = new Gson();
+    String json = gson.toJson(message);
+    byte[] return_json = json.getBytes();
+    return return_json;
+}
+
+/**
+ * This function is used to send out encoded timestep update to all registered instances the manager has on the managed instances map
+ * 
+ * @param message This time message is used to store current seq and timestep from the ambassador side
+ * @throws IOException
+ */
+public void onTimestepUpdate(InfrastructureTimeMessage message) throws IOException 
+{
+    if(this.manager.get_managedInstances().size() == 0)
+    {
+        throw new IllegalAccessError("There are no registered instances!");
```

Comment:
I'm not sure that this is necessary, and may actually be a problem. In this case, if this is called every timestep then any timestep where we don't have any infrastructure instances registered would mean we either crash the infrastructure ambassador or the whole MOSAIC RTI. For scenarios where we're not using infrastructure simulation that would be a problem as well as any case where - due to the time it takes to launch the software - the infrastructure isn't able to connect on exactly the first timestep of simulation.

Regardless, if the role of this function is to send out messages to registered instances, then if there are 0 registered instances it seems straightforward to send 0 messages. Might be worthwhile to keep it as a log statement though, so we can catch if there are times where infrastructure is slow to register or somethin.

Reply:
Removed the throw exception, added a log.info to state that there are no registered instances